extension/memorylimiter/README.md

```diff
@@ -0,0 +1,101 @@
+# Memory Limiter extension
+
+| Status                   |                   |
+|--------------------------|-------------------|
+| Stability                | [alpha]           |
+| Distributions            | [core], [contrib] |
+
+The memory limiter extension is used to prevent out of memory situations on
+the collector. Given that the amount and type of data the collector processes is
+environment specific and resource utilization of the collector is also dependent
+on the configured extensions, it is important to put checks in place regarding
+memory usage.
+ 
+The memory_limiter extension allows to perform periodic checks of memory
+usage if it exceeds defined limits will begin refusing data and forcing GC to reduce
+memory consumption.
+
+The memory_limiter uses soft and hard memory limits. Hard limit is always above or equal
+the soft limit.
+
+When the memory usage exceeds the soft limit the extension will enter the memory limited
+mode and will start refusing the data by returning errors to the preceding component.
+The preceding component should be normally a receiver.
```

Comment:
Is this description still accurate if this is an extension? The preceding step should be the client sending events to the receiver right?